src/components/calendar/calendar.js

```diff
@@ -686,4 +675,22 @@
 
     return monthBody;
   };
+
+
+  /**
+   * Gets an identifier for a date unique to the calendar instance for internal
+   * purposes. Not to be displayed.
+   * @param {Date} date
+   * @returns {string}
+   * @private
+   */
+  CalendarCtrl.prototype.getDateId_ = function (date) {
+    return [
+      'md',
+      this.$scope.$id,
```

Comment:
Instead of using the scope ID, can you just generate a unique ID in the constructor? If the component changes to not create a new scope I wouldn't want there to be a collision.